crates/engine/tree/src/tree/root.rs

```diff
@@ -438,6 +442,24 @@ where
     }
 }
 
+#[derive(Debug, Default)]
+struct DecodedProofs {
+    pub decoded_nodes: Vec<TrieNode>,
+}
+
+fn decode_proofs(multiproof: MultiProof) -> Result<DecodedProofs, SparseStateTrieError> {
+    let mut nodes: Vec<TrieNode> = vec![];
+    let account_subtree = multiproof.account_subtree.into_nodes_sorted();
+    let account_nodes = account_subtree.into_iter();
+
+    // Reveal the remaining proof nodes.
+    for (_path, bytes) in account_nodes {
+        let node = TrieNode::decode(&mut &bytes[..])?;
+        nodes.push(node);
+    }
+    Ok(DecodedProofs { decoded_nodes: nodes })
+}
```

Comment:
Instead of this, let's just start by creating a struct like `StorageMultiProof`:
https://github.com/paradigmxyz/reth/blob/905fd37bde758312364c73f28a7c6ba3f3349fb4/crates/trie/common/src/proofs.rs#L141

In trie-common, that uses `alloy_trie::DecodedProofNodes` instead of `ProofNodes`